Slack notice

dubb-b opened the pull request:
### What does this PR do?
Adds slack notice to failed PR's in channel `#jenkins-prod-pr`

### What issues does this PR fix or reference?
New Feature

### Previous Behavior
None

### New Behavior
Awesome notifications now!!!

### Commits signed with GPG?

Yes

twangboy committed:
Populate changes dictionary

Fix docs formatting
Clean up code
Use explicit function calls
Add comment about reboot status

twangboy committed:
Add ability to clear DNS entries on static DNS

Execution Module:
- Passing no DNS servers will clear the list.
- Update documentation
- Added a note about clearing DNS entries

State Module
- An empty list in dns_servers will clear the entires
- None in dns_servers will do nothing (default behavior)
- Update Documentation

terminalmage committed:
Fix parallel states with long ID dec or name

The cache files used for these are based on the state's tag, so longer
ID decs or names will cause errors when the filename's length exceeds
the max allowed by the kernel.

This fixes these errors by taking (up to) the first 32 chars of the
result of base64-encoding the tag, and using that as the parallel cache
filename.

terminalmage committed:
Remove/replace unnecessary setUp/tearDown

The setUp only needs to run at the beginning so it can be a setUpClass,
and the tearDown logic only applied to a single test and should not be
run after each test.

twangboy committed:
Only clear dns entries if you pass []

Add test=True support when clearing

twangboy committed:
Add/fix tests

Use named parameters where possible

Commit:
Merge branch '2017.7' into fix_49730

Commit:
Merge branch '2017.7' into fix_49730

terminalmage committed:
Add additional info to the assert to aid in troubleshooting

terminalmage committed:
Skip parallel states test on Windows until we can get it to work

terminalmage committed:
Explicitly import salt.utils.hashutils

The code works without this because at least one of the other modules
being imported also imports this. But explicitly importing it here will
keep us from getting bitten if the imports change elsewhere later on.

terminalmage committed:
Switch from base64 encoding to sha1 digest for unique cache filename

terminalmage committed:
Update test docstring to reflect switch to using SHA1

Commit:
Adding timeout to all pipelines so that the build aborts

Commit:
Updating syntax to be better and with new lines

dwoz committed:
Use utf-8 encoding for salt-runtests.log

twangboy committed:
Fix broken tests, add new tests

Commit:
Adding some changes for the status

Commit:
Adding slack notices for failed PRs

damon-atkins commented:
Try this out, but you will need to write the code I assume.  Given all tests need a working python you could use python to pull the info out of github.
https://api.github.com/repos/SaltStack/Salt/pulls/49831
Which will give you url for each reviewer "url": "https://api.github.com/users/gtmanfred",
Then in the message assume the github username and slack are the same and use @dubb-b in the slack message.

dubb-b commented:
@damon-atkins Are you in community slack? If so hit me up, lets talk more about this. I am just trying to get a slack message for any PR that fails into a slack message so that we can start to identify them. But this is just a stop gap for now. We want to add some sort of TICK stack to Jenkins so that we get metrics on all builds, branch,  PR etc.. Then we can group the data better. But I would love to hear any ideas you have. I've noticed you comment about this before. This process is one of our top priorities. And any feedback is welcome.

In your suggestion for improving this do you think adding more data is relevant? I could have a script that pulls the reviewer into the message. And quite a bit more if needed. @rallytime would this help at all?

I am checking out for the eve, although Monday feel free to hit me up.

Commit:
Merge pull request #49823 from dubb-b/timeout_addition

Adding timeout to all pipelines so that the build aborts

Commit:
Merge branch '2017.7' into fix_49730

Commit:
Merge pull request #49763 from twangboy/fix_49730

Add ability to clear DNS entries on static DNS

Commit:
Merge branch '2017.7' into testslogencoding

rallytime commented:
> Then in the message assume the github username and slack are the same

We can't really assume that. There's lots of people who don't have the same GitHub and Slack usernames.

@dubb-b It seems here that you've got more commits in this than intended - this looks like you've got the stuff that has already been merged in here (i.e. the timeouts for docs/lint) rather than _just_ the slack additions. Maybe this needs a rebase? 

We can definitely chat more about this on Monday, but I think the general implementation is a great place to start.

Commit:
Merge pull request #49777 from terminalmage/issue49738

Fix parallel states with long ID dec or name

Commit:
Merge branch '2017.7' into fix_49660

gtmanfred commented on .ci/docs:
```diff
@@ -45,6 +48,7 @@ pipeline {
                 description: 'The docs job has failed',
                 status: 'FAILURE',
                 context: "jenkins/pr/docs"
+            slackSend (channel: "#jenkins-prod-pr", color: '#FF0000', message: "FAILED: PR-Job: '${env.JOB_NAME} [${env.BUILD_NUMBER}]' (${env.BUILD_URL})" + "\n It's not denial. I'm just selective about the reality I accept.")
```
this indent looks wrong.

gtmanfred reviewed (changes requested):
A few cosmetic changes.

terminalmage committed:
Fix badly formatted versionadded directive

A space here will cause the Sphinx role to treat whatever is after the
comma as the comment for the directive, so it will be rendered with
different CSS and look weird. Also, this fixes one syntactically
incorrect versionadded which included a typo.

Commit:
Merge pull request #49848 from terminalmage/fix-version-rst-2017.7

Fix badly formatted versionadded directive (2017.7 branch)

dwoz committed:
Merge pull request #49828 from dwoz/testslogencoding

Use utf-8 encoding for salt-runtests.log

rallytime reviewed (commented):
Once this is rebased I can review this.

Paulo-Nunes committed:
Edit requisite documentation

Fixes #49836

Commit:
Merge branch '2017.7' into fix_49660

Commit:
Merge pull request #49858 from Paulo-Nunes/fix-requisite-doc

Edit requisite documentation

KaiSforza committed:
Actually catch the exception when we fail

This should actually cause builds to fail when any part in the try block
fails. We still delete instances and try junit rendering, but if things
fail then they get marked on github as failed.

gtmanfred committed:
Merge pull request #49870 from KaiSforza/ci_actually_fail

Actually catch the exception when we fail

dwoz committed:
Move vagrant to it's own group

Better support for ec2 on laptops that also have vagrant installed.

dwoz committed:
Use dwoz/winrm-fs for chunked downloads

Revert this after there is an upstream release

dwoz committed:
Merge pull request #49885 from dwoz/gemfile

Gemfile updates

Commit:
Merge pull request #49739 from twangboy/fix_49660

Populate changes dictionary on system.join_domain

twangboy committed:
Remove arrInstalled artifact

Commit:
Merge pull request #49899 from twangboy/fix_49892

Remove arrInstalled artifact

rallytime commented:
@dubb-b Where are we with this one?

Commit:
Updating formatting and the message

Commit:
Updating Merege conflict
